apps/src/lib/client/utils.rs

```diff
@@ -44,161 +44,177 @@ pub const NET_OTHER_ACCOUNTS_DIR: &str = "other";
-        fs::rename(
-            unpack_dir
-                .join(config::DEFAULT_BASE_DIR)
-                .join(config::global::FILENAME),
-            base_dir_full.join(config::global::FILENAME),
-        )
-        .await
-        .unwrap();
+            // Move the global config
+            fs::rename(
+                unpack_dir
+                    .join(config::DEFAULT_BASE_DIR)
+                    .join(config::global::FILENAME),
+                base_dir_full.join(config::global::FILENAME),
+            )
+                .await?;
 
-        // Remove the default dir
-        fs::remove_dir_all(unpack_dir.join(config::DEFAULT_BASE_DIR))
-            .await
-            .unwrap();
-    }
+            // Remove the default dir
+            fs::remove_dir_all(unpack_dir.join(config::DEFAULT_BASE_DIR))
+                .await?;
+        }
 
-    // Move wasm-dir and update config if it's non-default
-    if let Some(wasm_dir) = wasm_dir.as_ref() {
-        if wasm_dir.to_string_lossy() != config::DEFAULT_WASM_DIR {
-            tokio::fs::rename(
-                base_dir_full
-                    .join(chain_id.as_str())
-                    .join(config::DEFAULT_WASM_DIR),
-                base_dir_full.join(chain_id.as_str()).join(wasm_dir),
-            )
-            .await
-            .unwrap();
+        // Move wasm-dir and update config if it's non-default
+        if let Some(wasm_dir) = wasm_dir.as_ref() {
+            if wasm_dir.to_string_lossy() != config::DEFAULT_WASM_DIR {
+                tokio::fs::rename(
+                    base_dir_full
+                        .join(chain_id.as_str())
+                        .join(config::DEFAULT_WASM_DIR),
+                    base_dir_full.join(chain_id.as_str()).join(wasm_dir),
+                )
+                    .await?;
+
+                // Update the config
+                let wasm_dir = wasm_dir.clone();
+                let base_dir = base_dir.clone();
+                let chain_id = chain_id.clone();
+                tokio::task::spawn_blocking(move || {
+                    let mut config = Config::load(
+                        &base_dir,
+                        &chain_id,
+                        global_args.mode.clone(),
+                    );
+                    config.wasm_dir = wasm_dir;
 
-            // Update the config
-            let wasm_dir = wasm_dir.clone();
-            let base_dir = base_dir.clone();
-            let chain_id = chain_id.clone();
-            tokio::task::spawn_blocking(move || {
-                let mut config = Config::load(
-                    &base_dir,
-                    &chain_id,
-                    global_args.mode.clone(),
-                );
-                config.wasm_dir = wasm_dir;
-                config.write(&base_dir, &chain_id, true).unwrap();
-            })
-            .await
-            .unwrap();
+                    exit_if_error(config.write(&base_dir, &chain_id, true));
```

Comment:
NB: having to call an `exit_if_error` function here directly as we are running this block of code in another thread IIUC, though thinking it might be ok to run this block in the same thread as the rest of the function